go/worker/compute/executor/committee/node.go

```diff
@@ -790,9 +797,18 @@ func (n *Node) handleScheduleBatch(force bool) {
 		return
 	}
 
-	// Ask the scheduler to get us a scheduled batch.
+	// Ask the scheduler to get a batch of transactions for us and see if we should be proposing
+	// a new batch to other nodes.
 	batch := n.scheduler.GetBatch(force)
-	if len(batch) == 0 && (!force || len(roundResults.Messages) == 0) {
+	switch {
+	case len(batch) > 0:
+		// We have some transactions, schedule batch.
+	case len(roundResults.Messages) > 0:
```

Comment:
I think that if the batch is not ready yet, we should not schedule a batch unless batch flush timeout is reached (`force==true`), even if there are messages?

Reply:
Yeah makes sense, will change it back.